identity/common/src/main/java/io/camunda/identity/permissions/PermissionEnum.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.identity.permissions;
+
+public enum PermissionEnum {
+  CREATE_ALL,
+  READ_ALL,
+  UPDATE_ALL,
+  DELETE_ALL;
```

Comment:
Let's add a text representation for these roles. My proposal is:
- create all `*:create`
- read all `*:read`
- update all `*:update`
- delete all `*:delete`

Reply:
I had that in the first place and am happy you're bringing that up now. I was wondering where the system will benefit from that?

We will use the Java Enum everywhere in the backend code of our one-application, so the enum fields are the ones we see when working with Permissions in Java. 

In which place will it help to have the `*:create` etc. notations? 

Just to explain: my intention here is to have less code and adding those values to the enum would generate more code depending on where these representations should appear. If we can narrow that down, I'll be able to put it in the right place only :)

Reply:
Storing retrieving permissions in a specific format allows us to create a structure that is extendable. We will have a permission check on the API endpoints which would looks something like:

```
@HasPermission(CREATE_USERS)
```

But actually you need to check `@HasPermission(CREATE_USERS) or HasPermissions(CREATE_ALL)`, extending the permission concept to "full access for users" will then be `@HasPermission(CREATE_USERS) or HasPermissions(CREATE_ALL) or HasPermission(USERS_ALL)`, so each time all of the annotations have to be changed in the code. Having a structure like "*:create" allows us to extend the functionality by just extending the annotation, for example

```
@HasPermission('/users', 'write')
```
Can internally check if any of the required permissions are present

Reply:
Thanks for your reply! I've added the string values to the enum. Let's see how to expand that in the future then.